src/PAModel/Utility/FilePath.cs

```diff
@@ -30,6 +33,29 @@ public string ToPlatformPath()
             return Path.Combine(_pathSegments.Select(Utilities.EscapeFilename).ToArray());
         }
 
+        /// <summary>
+        /// Checks if the path length is more than MAX_PATH (260)
+        /// then it truncates the path, and adds a hash to the filename while retaining the file extension.
+        /// </summary>
+        /// <param name="path">Full path of the file.</param>
+        /// <returns></returns>
+        public static string ToValidPath(string path)
```

Comment:
>public static string ToValidPath(string path) [](http://example.com/codeflow?start=8&length=45)

stepping back, Every other tool must be hitting this problem ... is there a common solution?